test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2430,4 +2431,31 @@ public static <T extends Throwable> T expectThrows(Class<T> expectedType, Reques
             () -> builder.get().decRef() // dec ref if we unexpectedly fail to not leak transport response
         );
     }
+
+    /**
+     * Run {@code numberOfTasks} parallel tasks that were created by the given {@code taskFactory}. On of the tasks will be run on the
+     * calling thread, the rest will be run on a new thread.
+     * @param numberOfTasks number of tasks to run in parallel
+     * @param taskFactory task factory
+     */
+    public static void runInParallel(int numberOfTasks, IntFunction<Runnable> taskFactory) throws InterruptedException, ExecutionException {
```

Comment:
`IntFunction<Runnable>` is kinda awkward, could this be an `IntConsumer` instead?

Reply:
Right that's much nicer :) thanks!